browser/src/control/Control.UIManager.ts

```diff
@@ -1460,8 +1460,29 @@ class UIManager extends window.L.Control {
 				app.socket.sendMessage('uno .uno:SidebarHide');
 			}
 		}
+		this.updateReadonlyIndicator();
 	}
 
+	/**
+	 * Updates visibility of the Read-only badge/button in the top toolbar.
+	*/
+	updateReadonlyIndicator(): void {
+		const topReadonlyBtn: HTMLElement | null = document.getElementById('readonlyMode');
```

Comment:
can we remember the instance of a button and initialize it once?
```
if (!this.topReadOnlyButton) {
    this.topReadOnlyButton = ...;
    const label = this.topReadOnlyButton.query...;
    if (label) label.textContent = ....;
}
```
this saves us from DOM touching in middle of message queue processing